include/class.thread_actions.php

```diff
@@ -366,3 +366,42 @@ protected function trigger__post() {
     }
 }
 ThreadEntry::registerAction(/* trans */ 'Manage', 'TEA_ResendThreadEntry');
+
+class TEA_CreateTask extends ThreadEntryAction {
+    static $id = 'create_task';
+    static $name = /* trans */ 'Create Task';
+    static $icon = 'plus';
+
+    function isVisible() {
+        global $thisstaff;
+
+        return $thisstaff && $thisstaff->hasPerm(Task::PERM_CREATE, false);
+    }
+
+    function getJsStub() {
+        return sprintf(<<<JS
+var url = '%s';
+var redirect = $(this).data('redirect');
+$.dialog(url, [201], function(xhr, resp) {
+    if (!!redirect)
+        $.pjax({url: redirect, container:'#pjax-container'});
+    else
+        $.pjax({url: 'ajax.php/tickets/378/tasks', container: '#tasks_content', push: false});
```

Comment:
oops. Should the ticket-id be replaced here?

Reply:
Doh. Yessir. Thank ya 😉